docs/guide/ethereum-provider.md

```diff
@@ -112,83 +173,359 @@ params: [
+- Detect which Ethereum network the user is connected to
+- Get the user's Ethereum account(s)
+
+<<< @/docs/snippets/handleProvider.js
+
+## Experimental API
 
 ::: warning
-Use `ethereum.sendAsync({ method: 'eth_requestAccounts' }, callback)` instead.
+There is no guarantee that the methods and properties defined in this section will remain stable.
+Use it at your own risk.
 :::
 
-Requests to view the user's Ethereum address.
-Returns a promise of an array of hex-prefixed ethereum address strings.
+We expose some experimental, MetaMask-specific methods under the `ethereum._metamask` property.
+
+## Experimental Methods
+
+### ethereum.\_metamask.isApproved() (TO BE REMOVED)
+
+::: danger DANGER
+This will be removed in **Q3 2020**.
+:::
+
+```typescript
+ethereum._metamask.isApproved(): Promise<boolean>;
+```
+
+This method returns a `Promise` that resolves to a `boolean` indicating if the caller has access to user accounts.
 
-### ethereum.send(payload, callback) (DEPRECATED)
+### ethereum.\_metamask.isEnabled() (TO BE REMOVED)
+
+::: danger DANGER
+This will be removed in **Q3 2020**.
+:::
+
+```typescript
+ethereum._metamask.isEnabled(): boolean;
+```
+
+This method returns a `boolean` indicating if the caller has access to user accounts.
+
+### ethereum.\_metamask.isUnlocked()
+
+```typescript
+ethereum._metamask.isUnlocked(): Promise<boolean>;
+```
+
+This method returns a `Promise` that resolves to a `boolean` indicating if MetaMask is unlocked by the user.
+This is useful for knowing if MetaMask is unlocked in order to provide meaningful instructions to the user during onboarding.
+Note that this does not indicate if the user has exposed any accounts.
+
+## Legacy API
 
 ::: warning
-Use `ethereum.sendAsync()` instead.
+You should **never** rely on any of these methods, properties, or events in practice.
 :::
 
-See `ethereum.sendAsync`, directly below.
+Historically, the Ethereum provider API has been a mess.
+Here, we document the remains of this mess in case you see it used in the wild and wonder what's going on.
 
-### ethereum.on(eventName, callback)
+## Legacy Properties
 
-The provider supports listening for some events:
+### ethereum.autoRefreshOnNetworkChange (TO BE REMOVED)
 
-- `accountsChanged`, returns the currently available account(s).
-- `networkChanged`, returns the current network ID as a decimal string.
-- `chainIdChanged`, returns the current chain ID as a hexadecimal string.
+::: danger DANGER
+When `window.web3` is removed in **Q4 2020**, this property will also be removed.
 
-#### Example
+If you don't access `window.web3`, the value of this property will not affect the behavior of your application or MetaMask.
+
+As the consumer of this API, it is your responsbility to handle chain changes using the [`chainChanged` event](#chainChanged).
+We recommend reloading the page on `chainChange` or using [ethers](https://www.npmjs.com/package/ethers), which will do so for you.
+:::
+
+By default, this property is `true`.
```

Comment:
This isn't quite true. By default this isn't set at all, and we print a console warning asking the dapp to set it.

Reply:
No, we tell dapps to set it to `false` to silence its related warning, but it is set to `true` by us on initialization.

Reply:
Huh. Nevermind then. I remembered this incorrectly.